apps/vaporgui/VaporWidgets.h

```diff
@@ -0,0 +1,112 @@
+#ifndef VAPORWIDGETS_H 
+#define VAPORWIDGETS_H 
+
+#include <QWidget>
+#include <QDoubleValidator>
+
+#include "Combo.h"
+#include "ui_VPushButton.h"
+#include "ui_VComboBox.h"
+#include "ui_VCheckBox.h"
+#include "ui_VPathSelector.h"
+
+namespace Ui {
+class QSliderEdit;
+}
+
+
+class VPushButton: public QWidget, public Ui_VPushButton
+{
+    Q_OBJECT
+
+public:
+    VPushButton(
+        QWidget* parent, 
+        std::string labelText = "Label",
+        std::string buttonText = "Button"
+    );
+
+    void SetLabelText(  const std::string text );
+    void SetLabelText(  const QString text );
+    void SetButtonText( const std::string text );
+    void SetButtonText( const QString text );
+
+signals:
+    void _pressed();
+
+private slots:
+    void _buttonPressed();
+};
+
+
+
+class VComboBox : public QWidget, public Ui_VComboBox
+{
+    Q_OBJECT
+
+public:
+    VComboBox(
+        QWidget* parent,
+        std::string labelText = "Label"
+    );
+    void        SetLabelText(  const std::string text );
+    int         GetCurrentIndex() const;
+    std::string GetCurrentText() const;
+    void        AddOption( std::string option, int index=0 );
+    void        RemoveOption( int index );
+
+private slots:
+    void _userIndexChanged(int index);
+
+signals:
+    void _indexChanged(int index);
+};
+
+
+
+class VCheckBox : public QWidget, public Ui_VCheckBox
+{
+    Q_OBJECT
+
+public:
+    VCheckBox(
+        QWidget* parent,
+        std::string labelText = "Label"
+    );
+    void SetLabelText( std::string text );
```

Comment:
Same argument here regarding reference, e.g., `const std::string&`

Reply:
Got it.